server/etcdserver/v3_server.go

```diff
@@ -258,6 +258,9 @@ func (s *EtcdServer) Compact(ctx context.Context, r *pb.CompactionRequest) (*pb.
 }
 
 func (s *EtcdServer) LeaseGrant(ctx context.Context, r *pb.LeaseGrantRequest) (*pb.LeaseGrantResponse, error) {
+	if r.ID < int64(lease.NoLease) {
```

Comment:
```suggestion
	if r.ID <= int64(lease.NoLease) {
```

leaseID equal 0 represents no lease so it also should not be allowed in API
https://github.com/etcd-io/etcd/blob/170d9b9d73c8358043890332b7f4d7b27dbe7640/server/lease/lessor.go#L35